### PR TITLE
nodejs: update to 18.12.1

### PR DESCRIPTION
**Pull request by raedrizqie**

### CODE REVIEW

**Comment by Biswa96** [comment]
Thank you for this huge effort. Just curious about one thing. Is it possible to build v8 and node in separate packages? That may reduce the build time.

**Comment by Biswa96** [comment]
It's working

```
$ node
Welcome to Node.js v18.12.1.
Type ".help" for more information.
> console.log("Hello World!")
Hello World!
undefined
```

**Comment by raedrizqie** [comment]
> Thank you for this huge effort. Just curious about one thing. Is it possible to build v8 and node in separate packages? That may reduce the build time.

yes it is possible

File: mingw-w64-nodejs/0013-Revert-to-FHS-installation-paths-on-Windows-tests-no.patch [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+index 5acd6fb6..dc01687e 100755
+--- a/deps/npm/bin/npm
++++ b/deps/npm/bin/npm
+@@ -19,6 +19,9 @@
+ # kind of paths Node.js thinks it's using, typically win32 paths.
+ CLI_BASEDIR="$("$NODE_EXE" -p 'require("path").dirname(process.execPath)')"
+ NPM_CLI_JS="$CLI_BASEDIR/node_modules/npm/bin/npm-cli.js"
++if [ `uname` = 'MINGW*' ] ; then
++  NPM_CLI_JS="$CLI_BASEDIR/../lib/node_modules/npm/bin/npm-cli.js"
++fi
```
Comment (lazka) [inline mingw-w64-nodejs/0013-Revert-to-FHS-installation-paths-on-Windows-tests-no.patch:38]: @raedrizqie just wondering, did those "uname" checks come from you? I'm a bit confused how that is supposed to work :)

Comment (raedrizqie) [reply mingw-w64-nodejs/0013-Revert-to-FHS-installation-paths-on-Windows-tests-no.patch:38]: nope..I think its from the old patch.. i was just rebasing it

**Pull request merged**